ophyd/signal.py

```diff
@@ -648,6 +650,16 @@ def __init__(self, read_pv, *, string=False, auto_monitor=False, name=None,
         if timeout is DEFAULT_TIMEOUT:
             timeout = self.__default_timeout
         self._timeout = timeout
+        if write_timeout is DEFAULT_WRITE_TIMEOUT:
+            # This is very different than the connection and read timeouts
+            # above. It relates to how long an action takes to complete. Any
+            # default value we choose here is likely to cause problems---either
+            # by being too short and giving up too early on a lengthy action or
+            # being too long and delaying the report of a failure.
+            # The important thing it is that it is configurable at per-Signal
+            # level via the write_timeout parameter.
```

Comment:
Should note here or in a docstring that this write timeout is only invoked if the pv is already connected. A disconnected PV will be subject to the connection timeout instead. Even with the lengthy comment I don't think this is clear, before looking under the hood I was confused.

Reply:
Good note. The other timeouts needed detailed docstrings as well. Pushed.